src/app/data-planes/data/index.ts

```diff
@@ -78,29 +81,31 @@ export type MeshGatewayDataplane = PartialMeshGatewayDataplane & {
 }
 
 const DataplaneNetworking = {
-  fromObject(partialDataplaneNetworking: PartialDataplaneNetworking): DataplaneNetworking {
-    const { inbound, outbound, ...rest } = partialDataplaneNetworking
-
-    const inbounds: DataplaneInbound[] = (inbound ?? []).map((inbound) => {
-      // An inbound without a health property is considered healthy.
-      const health = { ready: !inbound.health || inbound.health.ready }
-      const addressPort = `${inbound.address ?? partialDataplaneNetworking.advertisedAddress ?? partialDataplaneNetworking.address}:${inbound.port}`
-      const serviceAddressPort = `${inbound.serviceAddress ?? inbound.address ?? partialDataplaneNetworking.address}:${inbound.servicePort ?? inbound.port}`
-
-      return {
-        ...inbound,
-        health,
-        addressPort,
-        serviceAddressPort,
-      }
-    })
+  fromObject(networking: PartialDataplaneNetworking): DataplaneNetworking {
+    // remove singular inbound/outbound to be replaced with plural versions
+    const { inbound, outbound, ...rest } = networking
 
-    const outbounds: DataplaneOutbound[] = outbound ?? []
+    const inbounds = Array.isArray(inbound) ? inbound : []
+    const outbounds = Array.isArray(outbound) ? outbound : []
 
     return {
       ...rest,
-      inbounds,
-      outbounds,
+      inbounds: inbounds.map((item) => {
+        return {
+          ...item,
+          // If a health property is unset the inbound is considered healthy
+          health: { ready: !isSet(item.health) ? true : item.health.ready },
```

Comment:
In 2.6 we also have a `State`: https://github.com/kumahq/kuma/blob/eb4b3d241742ce75950008b56daa58a3a9c402ad/api/mesh/v1alpha1/dataplane.proto#L162

Reply:
Ah perfect! I noticed that the other day but didn't realize how new it was 🎉 , much better!